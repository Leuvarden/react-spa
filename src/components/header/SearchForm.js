import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import uniqueId from 'lodash/uniqueId';

class SearchForm extends Component {
    render () {
        return ( 
            <Fragment>
                <div className="search-panel__form">
                    <label htmlFor="searchPanelInput">Find your movie</label>
                    {this.getInput()}
                </div>
        
                <div className="search-panel__filter">
                    <label id="filterDescription">Search by:</label>
                    {this.getFilterButtons()}
                    {this.getSearchButton()}
                </div>
            </Fragment>
        );
    }

    getInput() {
        return (
            <input 
                type="text" 
                id="searchPanelInput"
                onKeyDown={(e) => {
                    if (e.keyCode === 13) { //enter
                        this.getTerm();
                    }
                }}
            />
        );
    }

    getSearchButton() {
        return (
            <button 
                className="search-panel__button-search"
                onClick={() => this.getTerm() }
            >
            Search
            </button>
        );
    }

    getTerm() {
        let term = document.getElementById('searchPanelInput').value;
        this.props.setSearchTerm(term);
        this.props.updateResults(this.props.searchBy, term);
    }

    getFilterButtons() {
        const searches = ['title', 'genres'];

        return searches.map((el) => 
            <button
                key={uniqueId(el)}
                className={
                    `search-panel__button-filter 
                    ${ el === this.props.searchBy ? 'search-panel__button-filter_active' : ''}`
                }
                onClick={() => {
                    this.props.setSearchCriterion(el);
                }}
            >
                {el}
            </button>
    
        );
    }
}

export default SearchForm;

SearchForm.propTypes = {
    searchBy: PropTypes.string,
    setSearchTerm: PropTypes.func,
    setSearchCriterion: PropTypes.func,
    updateResults: PropTypes.func
};

