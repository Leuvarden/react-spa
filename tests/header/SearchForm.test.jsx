import React from 'react';
import SearchForm from './../../src/components/header/SearchForm';


/* eslint-disable no-undef */
describe ('SearchForm', () => {

    it('renders as expected', () => {
        const form = shallow(<SearchForm />);
    
        expect(form).toMatchSnapshot();
    });

});