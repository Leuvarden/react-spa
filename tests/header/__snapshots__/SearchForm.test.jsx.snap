// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders as expected 1`] = `
<React.Fragment>
  <div
    className="search-panel__form"
  >
    <label
      htmlFor="searchPanelInput"
    >
      Find your movie
    </label>
    <input
      id="searchPanelInput"
      onKeyDown={[Function]}
      type="text"
    />
  </div>
  <div
    className="search-panel__filter"
  >
    <label
      id="filterDescription"
    >
      Search by:
    </label>
    <button
      className="search-panel__button-filter 
                    "
      key="title1"
      onClick={[Function]}
    >
      title
    </button>
    <button
      className="search-panel__button-filter 
                    "
      key="genres2"
      onClick={[Function]}
    >
      genres
    </button>
    <button
      className="search-panel__button-search"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</React.Fragment>
`;
